### PR TITLE
Changes for RHDEVDOCS-1612-1602

### PR DESCRIPTION
**Pull request by nkakkar81**
Changes  made to doc as per the following two JIRAs: 
1. https://issues.redhat.com/browse/RHDEVDOCS-1612
2. https://issues.redhat.com/browse/RHDEVDOCS-1602

### CODE REVIEW

**Comment by nkakkar81** [comment]
@Preeticp @amitkrout @dharmit : Request you all to please review and provide comments. Thanks

File: cli_reference/openshift_developer_cli/understanding-odo.adoc [hunk 1]
```diff
@@ -30,11 +30,11 @@ Application::
 An application is a program designed for end users. An application consists of multiple microservices or components that work individually to build the entire application.
 Examples of applications: a video game, a media player, a web browser.
 Component::
-A component is a building block of an application. Typically it is a software package, a web service, a web resource, or a module that encapsulates a set of related functions or data. 
+A component is a set of kubernetes resources which host code or data. These resources run in a specific order for the overall software system to work. Each component can be run and deployed separately.
```
Comment (dharmit) [inline cli_reference/openshift_developer_cli/understanding-odo.adoc]: My understanding from Gorkem's definition is that the resources need to be running in order for the overall system to work. The definition doesn't seem to imply that these resources should be running in a specific order as mentioned here.

Besides, as a dev, I can say that we do start things in a certain order when a component is pushed to OpenShift cluster. But beyond that, the resources don't need to be running in a specific order for things to work.

**Review by dharmit — CHANGES_REQUESTED** [review]

File: cli_reference/openshift_developer_cli/understanding-odo.adoc (hunk 1 above)
Comment (kadel) [reply cli_reference/openshift_developer_cli/understanding-odo.adoc]:
```suggestion
A component is a set of kubernetes resources which host code or data.  Each component can be run and deployed separately.
```
"These resources run in a specific order for the overall software system to work." This sentence can be a little bit misleading. I would just remove it.

File: cli_reference/openshift_developer_cli/understanding-odo.adoc [hunk 2]
```diff
@@ -30,11 +30,11 @@ Application::
 An application is a program designed for end users. An application consists of multiple microservices or components that work individually to build the entire application.
 Examples of applications: a video game, a media player, a web browser.
 Component::
-A component is a building block of an application. Typically it is a software package, a web service, a web resource, or a module that encapsulates a set of related functions or data. 
+A component is a set of kubernetes resources which host code or data. Each component can be run and deployed separately.
```
Comment (nkakkar81) [inline cli_reference/openshift_developer_cli/understanding-odo.adoc]: @dharmit @kadel @Preeticp - Have made the suggested changes. Please review. Thanks

Comment (dharmit) [reply cli_reference/openshift_developer_cli/understanding-odo.adoc]: Looks good. Thanks @nkakkar81 !

Comment (Preeticp) [reply cli_reference/openshift_developer_cli/understanding-odo.adoc]: LGTM 
@nkakkar81 please remove the WIP, in the description right at the top, mention that the PR has been reviewed by SME and peer reviewed by the DevTools team. Post that, ping Andrea either here or in the team docs channel and ask for a review and merge, please.

Comment (nkakkar81) [reply cli_reference/openshift_developer_cli/understanding-odo.adoc]: @bergerhoffer - This PR has been reviewed by the SME, and the peer review is also completed by the DevTools Team. Request you to please review and merge ( if there are no comments). Thanks!

**Comment by nkakkar81** [comment]
@bergerhoffer - This PR has been reviewed by the SME, and the peer review is also completed by the DevTools Team. Request you to please review and merge ( if there are no comments). Thanks!

**Comment by bergerhoffer** [comment]
Hi @nkakkar81, I'd be happy to take a look. Can you (or @Preeticp) let me know what version(s) this applies for? And also whether this has been QE reviewed (or if it doesn't need it)?

File: cli_reference/openshift_developer_cli/understanding-odo.adoc (hunk 2 above)
Comment (bergerhoffer) [reply cli_reference/openshift_developer_cli/understanding-odo.adoc]: kubernetes -> Kubernetes

**Review by bergerhoffer — COMMENTED** [review]
Just one capitalization item to fix, otherwise LGTM! Once you fix that (and squash your commits), I can merge. Just let me know the applicable versions.

**Comment by Preeticp** [comment]
Thank you Andrea! This would be applicable to 4.2, 4.3, and 4.4

**Comment by nkakkar81** [comment]
Hello @bergerhoffer - I have made the edit, please review and merge. Thanks!!

File: cli_reference/openshift_developer_cli/creating-an-application-with-a-database.adoc [hunk 3]
```diff
@@ -12,7 +12,7 @@ This example describes how to deploy and connect a database to a front-end appli
 * `{odo-title}` is installed.
 * `oc` client is installed.
 * You have a running {product-title} cluster. Developers can use link:https://cloud.redhat.com/openshift/install/crc/installer-provisioned?intcmp=7013a000002CtetAAC[CodeReady Containers (CRC)] to deploy a local {product-title} cluster quickly.
-* Service Catalog is enabled.
+* xref:../../applications/service_brokers/installing-service-catalog.adoc#sb-installing-service-catalog[Service Catalog] is enabled.
```
Comment (amitkrout) [inline cli_reference/openshift_developer_cli/creating-an-application-with-a-database.adoc]: +1

**Comment by amitkrout** [comment]
@nkakkar81 Thanks for the pr.

Doc change looks good to me.

**Comment by Preeticp** [comment]
Hi Andrea, we now have QE ack too.

**Comment by bergerhoffer** [comment]
Thanks for the update, looks good to me now. Merging!

**Pull request merged**